odoo/addons/base/models/ir_model.py

```diff
@@ -263,9 +263,27 @@ def _check_order(self):
             stored_fields = set(
                 model.field_id.filtered('store').mapped('name') + models.MAGIC_COLUMNS
             )
+            related_fields = set(model.field_id.filtered('related').mapped('name'))
+
             order_fields = RE_ORDER_FIELDS.findall(model.order)
+            def check_related(mname, fname):
+                ir_field = self.env["ir.model.fields"].search([('model', '=', mname), ('name', '=', fname)])
+                parts = list(reversed(ir_field.related.split(".")))
+                while parts:
+                    if not mname:
+                        return False
+                    part = parts.pop()
+                    ir_field = self.env["ir.model.fields"].search([('model', '=', mname), ('name', '=', part)])
+                    if not ir_field:
+                        return False
+                    if ir_field.related:
+                        parts.extend(reversed(ir_field.related.split(".")))
+                        continue
+                    mname = ir_field.relation
+                return ir_field.store
+
             for field in order_fields:
-                if field not in stored_fields:
+                if field not in stored_fields and not (field in related_fields and check_related(self.model, field)):
```

Comment:
Something like that ?

```suggestion
                if (
                     field not in stored_fields and 
                     (
                         field not in self.env[model.model] or  # field doesn't exist yet ? is it possible ? 
                         (self.env[model.model]._fields[field].inherited and not self.env[model.model]._fields[field].base_field.store)
                 ):
```

Reply:
It's still wrong... WIP